includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -871,6 +872,19 @@ private function is_product_supported() {
 		return apply_filters( 'wcpay_payment_request_is_product_supported', $is_supported, $product );
 	}
 
+	/**
+	 * Determine wether to filter the cart needs shipping address.
+	 *
+	 * @param boolean $needs_shipping_address Whether the cart needs a shipping address.
+	 */
+	public function filter_cart_needs_shipping_address( $needs_shipping_address ) {
+		if ( $this->has_subscription_product() && $this->get_shipping_method_count( true, true ) === 0 ) {
```

Comment:
```suggestion
		if ( $this->has_subscription_product() && $this->get_shipping_method_count() === 0 ) {
```
if method is only used here and thus always with `[true, true]` pair, then maybe we could default to `true` for these parameters in the `get_shipping_method_count`'s definition?

UPDATE: this comment might become redundant because of other comments below which (if agreed upon & applied) will lead to the removal of `get_shipping_method_count`